### ISSUE TITLE
Error Installing Nvidia Apex

### ISSUE THREAD

**Issue by akshatchhabra** [akshatchhabra/TransLRPL#7]
```
Processing /work/pi_adrozdov_umass_edu/akshatchhabr_umass_edu/TransCoder_Test/CodeGen/apex
  Running command python setup.py egg_info
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/work/pi_adrozdov_umass_edu/akshatchhabr_umass_edu/TransCoder_Test/CodeGen/apex/setup.py", line 130, in <module>
      _, bare_metal_version = get_cuda_bare_metal_version(CUDA_HOME)
    File "/work/pi_adrozdov_umass_edu/akshatchhabr_umass_edu/TransCoder_Test/CodeGen/apex/setup.py", line 17, in get_cuda_bare_metal_version
      raw_output = subprocess.check_output([cuda_dir + "/bin/nvcc", "-V"], universal_newlines=True)
  TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'


  torch.__version__  = 1.9.0+cu102


  Preparing metadata (setup.py) ... error
WARNING: Discarding file:///work/pi_adrozdov_umass_edu/akshatchhabr_umass_edu/TransCoder_Test/CodeGen/apex. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by vedasreeb29** [comment]
After this error was resolved by using the cuda from unity, another issue occurred with gcc version
In file included from /modules/apps/cuda/10.1.243/include/cuda_runtime.h:83,
         from <command-line>:
/modules/apps/cuda/10.1.243/include/crt/host_config.h:138:2: error: #error -- unsupported GNU version! gcc versions later than 8 are not supported!
 138 | #error -- unsupported GNU version! gcc versions later than 8 are not supported!
   | ^~~~~
error: command '/modules/apps/cuda/10.1.243/bin/nvcc' failed with exit status 1

**Comment by vedasreeb29** [comment]
Finally, the errors got resolved by installing newer versions of pytorch, cuda and using Python 3.7.6 as mentioned in https://github.com/NVIDIA/apex/issues/1598#issuecomment-1445893243

pip install torch==1.13.1+cu117 torchvision==0.14.1+cu117 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/cu117 

And using the below commands for apex installation worked 

git clone https://github.com/NVIDIA/apex
cd apex
python setup.py install --cuda_ext
